WooCommerce/Classes/ViewRelated/Orders/Order Details/Shipping Labels/Create Shipping Label Form/Package Details/ShippingLabelPackageDetailsViewModel.swift

```diff
@@ -104,6 +123,44 @@ final class ShippingLabelPackageDetailsViewModel: ObservableObject {
     }
 }
 
+// MARK: - Package Selection
+extension ShippingLabelPackageDetailsViewModel {
+    func didSelectPackage(_ id: String?) {
+        guard id != nil else { return }
+        guard let packagesResponse = packagesResponse else {
+            return
+        }
+
+        for customPackage in packagesResponse.customPackages {
+                if customPackage.title == id {
+                    selectedCustomPackage = customPackage
+                    selectedPredefinedPackage = nil
+                    return
+                }
+            }
+        for option in packagesResponse.predefinedOptions {
+                for predefinedPackage in option.predefinedPackages {
+                    if predefinedPackage.id == id {
+                        selectedCustomPackage = nil
+                        selectedPredefinedPackage = predefinedPackage
+                        return
+                    }
+                }
+            }
+        }
```

Comment:
How do you feel about abstracting these two `for` loops into functions. I think it will help a lot with readability!

Reply:
Done 38c0ec317be9ddbfaebcefc614d4d08e27e419f9 👍